src/style-spec/validate/validate_expression.js

```diff
@@ -18,5 +18,12 @@ export default function validateExpression(options: any) {
         return [new ValidationError(options.key, options.value, 'Invalid data expression for "text-font". Output values must be contained as literals within the expression.')];
     }
 
+    if (options.expressionContext === 'property' && expression.value.kind === 'camera' && options.propertyKey.match(/(-pattern)|(-dasharray)/)) {
```

Comment:
Would it be slightly clearer to explicitly list all the properties this applies to instead of using a regular expression?

Reply:
It's another case where #6389 (specifically https://github.com/mapbox/mapbox-gl-js/issues/6389#issuecomment-376245679) would help.